### PR TITLE
nixos/v2raya: fix nftables support

### LINKED ISSUE #294091: The v2raya service doesn't allow v2rayA to discover nftables
### Describe the bug

On Arch Linux and NixOS, both with nftables installed, enabling the v2rayA **systemd service** with tproxy enabled will produce different nft rulesets. On Arch Linux, v2rayA creates a dedicated table for it, as shown by `nft list ruleset`:

    table inet v2raya {
            set whitelist {
                    type ipv4_addr
                    flags interval
                    auto-merge
                    elements = { 0.0.0.0, 10.0.0.0/8,
                                 100.64.0.0/10, 127.0.0.0/8,
    ...

On NixOS, however, if v2rayA is started by `v2raya.service`, the `v2raya` table is not present, and the rules with the TPROXY target are found in the `mangle` table instead:

    table ip mangle {
            ...
            chain TP_PRE {
                    ...
                    ip protocol tcp meta mark & 0x000000c0 == 0x00000040 counter packets 196 bytes 33331 xt target "TPROXY"
                    ip protocol udp meta mark & 0x000000c0 == 0x00000040 counter packets 72 bytes 5690 xt target "TPROXY"
            }
    ...

This is because, currently, https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/networking/v2raya.nix only adds `iptables` to the service `PATH`:

```nix
path = with pkgs; [ iptables bash iproute2 ]; # required by v2rayA TProxy functionality
```

However, v2rayA only consider nftables as supported if it finds the `nft` command:

https://github.com/v2rayA/v2rayA/blob/d8c97984206c97fb0da7c62b67906747e11d7327/service/core/iptables/utils.go#L64-L91

So I guess it just falls back to some legacy method of setting the forwarding rules.

In fact, v2rayA creates the same ruleset on NixOS after enabling `networking.nftables` if `v2rayA` is started directly from the command line instead of being started as a systemd service.

### Steps To Reproduce
1. Set `services.v2raya.enable = true;` and `networking.nftables.enable = true;`
2. In v2rayA's panel, enable transparent proxy with tproxy
3. List the the rule sets: `nft list ruleset`

### Expected behavior

There should be a table called `v2raya` under `inet`.

### Proposed Solution

I suggest modifying this line in `v2raya.nix`:

```diff
-path = with pkgs; [ iptables bash iproute2 ];
+path = with pkgs; [ iptables bash iproute2 ] ++ lib.optional nftablesEnabled nftables;
```

### Notify maintainers

CC @Eliot00  

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by LostAttractor**
## Description of changes
fix #294091 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/services/networking/v2raya.nix [hunk 1]
```diff
@@ -42,7 +42,7 @@ with lib;
         };
 
         wantedBy = [ "multi-user.target" ];
-        path = with pkgs; [ iptables bash iproute2 ]; # required by v2rayA TProxy functionality
+        path = with pkgs; [ iptables bash iproute2 ] ++ lib.optional nftablesEnabled nftables; # required by v2rayA TProxy functionality
```
Comment (Aleksanaa) [inline nixos/modules/services/networking/v2raya.nix:45]: The use of with here is not very good. Also, should it be `optionals`?

**Review by Aleksanaa — COMMENTED** [review]
The logic of using iptables and nftables in this module is a bit confusing. Some parts are a choice between the two, but in other parts `nftables` is optionally added. Could you tidy it at the same time?

**Comment by LostAttractor** [comment]
> The logic of using iptables and nftables in this module is a bit confusing. Some parts are a choice between the two, but in other parts `nftables` is optionally added. Could you tidy it at the same time?

It seems that v2raya cannot run without iptables as a dependency when using nftables

**Review by Aleksanaa — APPROVED** [review]
Tested

**Pull request merged**